### PR TITLE
Provide unlimited abstract length option

### PR DESCRIPTION
**Pull request by liang2kl**
It is useful to provide an option for users who want to publish the whole article, to be read directly on rss readers.

- Set default option for `abstract_chars_count` to `None`
- Returns full text when `abstract_chars_count` is `None`
- Change `on_page_markdown` to `on_page_content` to retrieve final html content for rss description

### CODE REVIEW

**Comment by Guts** [comment]
Thanks for your proposal.

> It is useful to provide an option for users who want to publish the whole article, to be read directly on rss readers.

Agreed!

> Set default option for abstract_chars_count to None

I prefer to keep it as an option to avoid a breaking change for existing installations.

> Returns full text when abstract_chars_count is None

Good move. Maybe when `abstract_chars_count==-1`, also to avoid a breaking change for existing installations.

> Change on_page_markdown to on_page_content to retrieve final html content for rss description

Are you totally sure thath it's not breaking something? If you take the full content, it's ok and easy-to-go. But if you specify a length, you'll need to manage a valid truncated HTML and it's not so simple.

Regarding failing tests, it's on [mkdocs-minify-plugin --> jsmin side](https://github.com/byrnereese/mkdocs-minify-plugin/issues/15). I'm monitoring the upstream PR.

**Comment by liang2kl** [comment]
> Maybe when abstract_chars_count==-1, also to avoid a breaking change for existing installations.

Agreed. I will update shortly.

> But if you specify a length, you'll need to manage a valid truncated HTML and it's not so simple.

In my pr, if the length is specified, the Markdown string, instead of HTML, will be used (which is the same as before, as on_page_markdown will not provide html string at all), so it should be fine.

**Comment by Guts** [comment]
> In my pr, if the length is specified, the Markdown string, instead of HTML, will be used (which is the same as before, as on_page_markdown will not provide html string at all), so it should be fine.

Great. I'll give it a closer look in the next days.

In the meanwhile, could you include the documentation update please?

And check if you follow the contributing guidelines. Thanks!

**Comment by liang2kl** [comment]
The docs have been updated.

File: mkdocs_rss_plugin/util.py [hunk 1]
```diff
@@ -258,31 +258,34 @@ def get_date_from_meta(
...
+        if description and chars_count != None:
+            return description
+        # If chars count is unlimited, use the html content
+        elif in_page.content and chars_count == None:
+            if chars_count == None or len(in_page.content) < chars_count:
+                return in_page.content[:chars_count]
+        # Use markdown
         elif in_page.markdown:
-            if len(in_page.markdown) < chars_count:
+            if chars_count == None or len(in_page.markdown) < chars_count:
```
Comment (Guts) [inline mkdocs_rss_plugin/util.py:288]: Condition should be `== -1 `, no?

File: mkdocs_rss_plugin/util.py [hunk 2]
```diff
@@ -258,31 +258,34 @@ def get_date_from_meta(
...
 
-    def get_description_or_abstract(self, in_page: Page, chars_count: int = 150) -> str:
+    def get_description_or_abstract(self, in_page: Page, chars_count: int = 160) -> str:
         """Returns description from page meta. If it doesn't exist, use the \
         {chars_count} first characters from page content (in markdown).
 
         :param Page in_page: page to look at
         :param int chars_count: if page.meta.description is not set, number of chars \
-        of the content to use. Defaults to: 150 - optional
+        of the content to use. Defaults to: None (unlimited) - optional
```
Comment (Guts) [inline mkdocs_rss_plugin/util.py:267]: Also update docstring

**Review by Guts — CHANGES_REQUESTED** [review]

File: mkdocs_rss_plugin/util.py (hunk 1 above)
Comment (liang2kl) [reply mkdocs_rss_plugin/util.py:288]: No, I set it to None if <0 to make it convenient for string slicing, please see lines above.

Comment (liang2kl) [reply mkdocs_rss_plugin/util.py:288]: `in_page.markdown[:None]` will result in the whole string

File: mkdocs_rss_plugin/util.py (hunk 2 above)
Comment (liang2kl) [reply mkdocs_rss_plugin/util.py]: Updated : )

**Review by Guts — APPROVED** [review]

**Comment by Guts** [comment]
Hi @liang2kl,

I've just reviewed it deeper and I've got two concerns:

- the output HTML should be included into a CDATA section since HTML into XML is not a straight forward thing (see: https://www.w3schools.com/XML/dom_cdatasection.asp), no?
- the embedded HTML is not valid because main tags are missing: `<html><body></body></html>`. Is that the behavior you want?

**Comment by liang2kl** [comment]
I don't have much understanding about the standards, but it seems that in your previous implementation, you deal it with the same way, generating html content without head tag, etc. and just embedding the html into xml?

**Comment by Guts** [comment]
Now deployed as part of the version 0.18.0:

- release: https://github.com/Guts/mkdocs-rss-plugin/releases/tag/0.18.0
- changelog: https://github.com/Guts/mkdocs-rss-plugin/blob/main/CHANGELOG.md#0180---2021-09-20

**Pull request merged**